tests/foreman/api/test_host.py

```diff
@@ -91,3 +93,244 @@ def test_positive_update_owner_type(self):
                 host.owner_type = owner_type
                 host = host.update(['owner_type'])
                 self.assertEqual(host.owner_type, owner_type)
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_name(self):
+        """Create a host with a random valid name
+
+        @feature: Hosts
+
+        @assert: A host is created and the name matches
+        """
+        for name in valid_hosts_list():
+            with self.subTest(name):
+                host = entities.Host()
+                host.create_missing()
+                host.name = name
+                host = host.create(create_missing=False)
```

Comment:
That was done on purpose. Take a look at [Host.create_missing()](https://github.com/SatelliteQE/nailgun/blob/master/nailgun/entities.py#L2031-L2093). Rules for creating missing entities for hosts are too complex and `create_missing()` works only for new instances of host, which do not have any instance attributes set. More information is in [Host.create_missing()](https://github.com/SatelliteQE/nailgun/blob/master/nailgun/entities.py#L2032-L2056) docstring. That's why we're creating dependencies at first, then updating required fields for future host and only then actually creating a host. Btw, same workaround is also used in CLI hosts tests.